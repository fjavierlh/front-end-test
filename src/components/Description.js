import React from 'react';
import styled from 'styled-components';

export const Description = ({ product }) => {
  const {
    brand,
    model,
    price,
    cpu,
    ram,
    os,
    battery,
    primaryCamera,
    secondaryCamera,
    displayResolution,
    dimensions,
    weight,
    displayType,
    displaySize,
    chipset,
    externalMemory,
    internalMemory,
    nfc,
    usb,
  } = { ...product };
  return (
    <DescriptionStyled>
      <div className="description">
        <dl className="description-left">
          {brand && (
            <>
              <dt>Brand</dt>
              <dd>{brand}</dd>
            </>
          )}
          {model && (
            <>
              <dt>Model</dt>
              <dd>{model}</dd>
            </>
          )}
          {price && (
            <>
              <dt>Price</dt>
              <dd>{price}â‚¬</dd>
            </>
          )}
          {cpu && (
            <>
              <dt>CPU</dt>
              <dd>{cpu}</dd>
            </>
          )}
          {ram && (
            <>
              <dt>RAM</dt>
              <dd>{ram}</dd>
            </>
          )}
          {os && (
            <>
              <dt>OS</dt>
              <dd>{os}</dd>
            </>
          )}
          {displayResolution && (
            <>
              <dt>Display Resolution</dt>
              <dd>{displayResolution}</dd>
            </>
          )}
          {battery && (
            <>
              <dt>Battery</dt>
              <dd>{battery}</dd>
            </>
          )}
          {dimensions && (
            <>
              <dt>Dimensions</dt>
              <dd>{dimensions}</dd>
            </>
          )}
          {weight && (
            <>
              <dt>Weight</dt>
              <dd>{weight}</dd>
            </>
          )}
        </dl>
        <dl className="description-left">
          {(primaryCamera || secondaryCamera) && (
            <>
              <dt>Cameras</dt>
              <dd>
                {primaryCamera && (
                  <>
                    <dt>Primary Camera</dt>
                    <dd>{primaryCamera}</dd>
                  </>
                )}
                {secondaryCamera && (
                  <>
                    <dt>Secondary Camera</dt>
                    <dd>{secondaryCamera}</dd>
                  </>
                )}
              </dd>
            </>
          )}

          {displayType && (
            <>
              <dt>Display Type</dt>
              <dd>{displayType}</dd>
            </>
          )}

          {displaySize && (
            <>
              <dt>Display Size</dt>
              <dd> {displaySize}</dd>
            </>
          )}

          {chipset && (
            <>
              <dt>Chipset</dt>
              <dd>{chipset}</dd>
            </>
          )}
          {externalMemory && (
            <>
              <dt>External Memory</dt>
              <dd>{externalMemory}</dd>
            </>
          )}
          {internalMemory && (
            <>
              <dt>Internal Memory</dt>
              <dd>{internalMemory}</dd>
            </>
          )}

          {nfc && (
            <>
              <dt>NFC</dt>
              <dd>{nfc}</dd>
            </>
          )}
          {usb && (
            <>
              <dt>USB</dt>
              <dd>{usb}</dd>
            </>
          )}
        </dl>
      </div>
    </DescriptionStyled>
  );
};

const DescriptionStyled = styled.article`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  dl {
    margin: 0;
  }

  dt {
    text-transform: uppercase;
    font-size: 0.9em;
    font-weight: bold;
    margin-top: 0.5rem;
    margin-top: 0.8rem;
  }

  dd > dt {
    font-size: 0.8em;
    margin-left: 2em;
    margin-bottom: 0.5rem;
  }
  dd > dd {
    font-size: 0.8em;
    margin-left: 2em;
  }

  dd {
    font-size: 0.9rem;
    margin: 0;
    font-weight: lighter;
  }

  & > .description {
    display: flex;
    gap: 1.5em;

    justify-content: space-evenly;
  }
`;
