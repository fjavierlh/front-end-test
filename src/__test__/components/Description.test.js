import { render, screen } from '@testing-library/react';
import Description from '../../components/Description';
import { singleProduct } from './../../../fixtures/singleProduct.fixture';

describe('Description', () => {
  test('should render Description', () => {
    render(<Description product={singleProduct} />);
  });

  test('should match snapshot Description when receive product by props', () => {
    const { container } = render(<Description product={singleProduct} />);
    expect(container).toMatchSnapshot();
  });

  test('should render product properties', () => {
    render(<Description product={singleProduct} />);
    expect(screen.queryByText(/brand/i)).toBeInTheDocument();
    expect(screen.queryByText(/model/i)).toBeInTheDocument();
    expect(screen.queryByText(/price/i)).toBeInTheDocument();
    expect(screen.queryByText(/cpu/i)).toBeInTheDocument();
    expect(screen.queryByText(/ram/i)).toBeInTheDocument();
    expect(screen.queryByText(/^os$/i)).toBeInTheDocument();
    expect(screen.queryByText(/display resolution/i)).toBeInTheDocument();
    expect(screen.queryByText(/battery/i)).toBeInTheDocument();
    expect(screen.queryByText(/cameras/i)).toBeInTheDocument();
    expect(screen.queryByText(/display type/i)).toBeInTheDocument();
    expect(screen.queryByText(/display size/i)).toBeInTheDocument();
    expect(screen.queryByText(/chipset/i)).toBeInTheDocument();
    expect(screen.queryByText(/external memory/i)).toBeInTheDocument();
    expect(screen.queryByText(/internal memory/i)).toBeInTheDocument();
    expect(screen.queryByText(/^usb$/i)).toBeInTheDocument();
  });
});
