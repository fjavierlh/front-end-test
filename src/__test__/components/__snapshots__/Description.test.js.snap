// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should match snapshot Description when receive product by props 1`] = `
<div>
  <article
    class="sc-bdvvtL iwcPrF"
  >
    <div
      class="description"
    >
      <dl
        class="description-left"
      >
        <dt>
          Brand
        </dt>
        <dd>
          Acer
        </dd>
        <dt>
          Model
        </dt>
        <dd>
          Iconia Talk S
        </dd>
        <dt>
          Price
        </dt>
        <dd>
          170
          â‚¬
        </dd>
        <dt>
          CPU
        </dt>
        <dd>
          Quad-core 1.3 GHz Cortex-A53
        </dd>
        <dt>
          RAM
        </dt>
        <dd>
          2 GB 
        </dd>
        <dt>
          OS
        </dt>
        <dd>
          Android 6.0 (Marshmallow)
        </dd>
        <dt>
          Display Resolution
        </dt>
        <dd>
          7.0 inches (~69.8% screen-to-body ratio)
        </dd>
        <dt>
          Battery
        </dt>
        <dd>
          Non-removable Li-Ion 3400 mAh  (12.92 Wh)
        </dd>
        <dt>
          Weight
        </dt>
        <dd>
          260
           gr
        </dd>
      </dl>
      <dl
        class="description-left"
      >
        <dt>
          Cameras
        </dt>
        <dl>
          <dt>
            Primary Camera
          </dt>
          <dd>
            13 MP
            autofocus
          </dd>
        </dl>
        <dt>
          Display Type
        </dt>
        <dd>
          IPS LCD capacitive touchscreen  16M colors
        </dd>
        <dt>
          Display Size
        </dt>
        <dd>
           
          720 x 1280 pixels (~210 ppi pixel density)
        </dd>
        <dt>
          Chipset
        </dt>
        <dd>
          Mediatek MT8735
        </dd>
        <dt>
          External Memory
        </dt>
        <dd>
          microSD  up to 128 GB (dedicated slot)
        </dd>
        <dt>
          Internal Memory
        </dt>
        <dd>
          16 GB
          32 GB
        </dd>
        
        <dt>
          USB
        </dt>
        <dd>
          microUSB 2.0
        </dd>
      </dl>
    </div>
  </article>
</div>
`;
